libcudacxx/include/cuda/__iterator/strided_iterator.h

```diff
@@ -0,0 +1,320 @@
+
+  //! NOTE: _Iter must be default initializable because it is a random_access_iterator and thereby semiregular
+  //!       _Stride must be integer-like or integral_constant_like which requires default constructability
+  _CCCL_EXEC_CHECK_DISABLE
+  _CCCL_HIDE_FROM_ABI strided_iterator() = default;
+
+  // We want to avoid constructing a strided_iterator with a value constructed __integer like__ stride, because that
+  // would value construct to 0 and incrementing the iterator would do nothing.
+  _CCCL_EXEC_CHECK_DISABLE
+  _CCCL_TEMPLATE(class _Stride2 = _Stride)
+  _CCCL_REQUIRES((!_CUDA_VSTD::__integer_like<_Stride2>) )
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr explicit strided_iterator(_Iter __iter) noexcept(
+    _CUDA_VSTD::is_nothrow_move_constructible_v<_Iter> && _CUDA_VSTD::is_nothrow_default_constructible_v<_Stride2>)
+      : __iter_(_CUDA_VSTD::move(__iter))
+      , __stride_()
+  {}
+
+  _CCCL_EXEC_CHECK_DISABLE
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr explicit strided_iterator(_Iter __iter, _Stride __stride) noexcept(
+    _CUDA_VSTD::is_nothrow_move_constructible_v<_Iter> && _CUDA_VSTD::is_nothrow_move_constructible_v<_Stride>)
+      : __iter_(_CUDA_VSTD::move(__iter))
+      , __stride_(_CUDA_VSTD::move(__stride))
+  {}
+
+  //! @brief Returns a const reference to the iterator stored in this \p transform_iterator
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr const _Iter& base() const& noexcept
+  {
+    return __iter_;
+  }
+
+  //! @brief Extracts the iterator stored in this \p transform_iterator
+  _CCCL_EXEC_CHECK_DISABLE
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr _Iter
+  base() && noexcept(_CUDA_VSTD::is_nothrow_move_constructible_v<_Iter>)
+  {
+    return _CUDA_VSTD::move(__iter_);
+  }
+
+  _CCCL_EXEC_CHECK_DISABLE
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr difference_type stride() const
+    noexcept(noexcept(static_cast<difference_type>(_CUDA_VSTD::__de_ice(_CUDA_VSTD::declval<const _Stride&>()))))
+  {
+    return static_cast<difference_type>(_CUDA_VSTD::__de_ice(__stride_));
+  }
+
+  _CCCL_EXEC_CHECK_DISABLE
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr decltype(auto) operator*() noexcept(noexcept(*__iter_))
+  {
+    return *__iter_;
+  }
+
+  _CCCL_EXEC_CHECK_DISABLE
+  _CCCL_TEMPLATE(class _Iter2 = _Iter)
+  _CCCL_REQUIRES(_CUDA_VSTD::__dereferenceable<const _Iter2>)
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr decltype(auto) operator*() const noexcept(noexcept(*__iter_))
+  {
+    return *__iter_;
+  }
+
+  _CCCL_EXEC_CHECK_DISABLE
+  [[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr decltype(auto) operator[](difference_type __n) const
+    noexcept(noexcept(*(__iter_ + __n)))
+  {
+    return *(__iter_ + __n);
+  }
+
+  _CCCL_EXEC_CHECK_DISABLE
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr strided_iterator&
+  operator++() noexcept(noexcept(__iter_ += static_cast<difference_type>(_CUDA_VSTD::__de_ice(__stride_))))
+  {
+    __iter_ += stride();
```

Comment:
Please unify the expression to `noexcept` with the function body:
```suggestion
  operator++() noexcept(noexcept(__iter_ += stride()))
  {
    __iter_ += stride();
```
Applies more often below.

Reply:
I had that before, but it breaks older gcc :(

Reply:
I slightly reworked it to be simpler